src/engraving/libmscore/stem.cpp

```diff
@@ -120,8 +120,11 @@ void Stem::layout()
     double lineX = isTabStaff ? 0.0 : _up * lineWidthCorrection;
     m_line.setLine(lineX, y1, lineX, y2);
 
+    // HACK: if there is a beam, extend the bounding box of the stem (NOT the stem itself) by half beam width.
+    // This way the bbox of the stem covers also the beam position. Hugely helps with slur collisions.
+    double beamCorrection = chord()->beam() ? _up * score()->styleMM(Sid::beamWidth) / 2 : 0.0;
     // compute line and bounding rectangle
-    RectF rect(m_line.p1(), m_line.p2());
+    RectF rect(m_line.p1(), m_line.p2() + PointF(0.0, beamCorrection));
```

Comment:
The way I understand it (but please correct me if I'm wrong), is that beams are not added to the shapes, but only to the skyline. The reason is that shapes are grouped segment by segment, but beams span multiple segments so you can't add them to any specific segment shape.
See for instance, when toggling the "Show segment shapes" diagnostic, beam shapes aren't included:
<img width="1366" alt="image" src="https://user-images.githubusercontent.com/93707756/186354641-3fba0a61-eb49-44b9-b34a-7f5c034c2b87.png">
but they are included in the skyline:
<img width="1228" alt="image" src="https://user-images.githubusercontent.com/93707756/186355024-abf8f1fb-2ac5-4885-8ca3-7810ed15c727.png">
You may argue that what we should be doing for slur collisions is looking at the skyline then, not at the shapes, and I'd completely agree and that's my plan for the future. But that requires an overhaul of the vertical spacing system (not too different than what I did with horizontal spacing), so it's not something which we can't do in the short term.

About doing this stem-hack in the slur collision code: a) I think it'd be uglier, and b) I've realized that this stem-hack turns out super useful also in other situations, for example:
<img width="1260" alt="image" src="https://user-images.githubusercontent.com/93707756/186357762-d1dc3ad4-c496-4353-af5f-1230b7e406e2.png">
(correct)
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/93707756/186357941-793090de-ae3f-490e-9707-ac19c312a926.png">
(wrong: sharp collides with the beam).
This is a horizontal spacing problem that has been **torturing** me and @oktophonie since forever and this simple stem-hack finally solves it :D


(About steep beams: their shape is also approximated by steps, so the stem-hack actually gives a more accurate result ;) ) 
<img width="350" alt="image" src="https://user-images.githubusercontent.com/93707756/186356669-89a7acbc-32f9-41a7-bd2d-b6ad9fe7e7b5.png">

Reply:
(The only apparent side-effect of this stem-hack is in the positioning of tuplet numbers, as shown in the vtests. But that's something quickly fixable.)

Reply:
Fair enough. Indeed, it looks like this hack is the best thing we can do for now.

And I can absolutely understand that the tucking-vs-beam problem would be _very_ difficult to solve... That would open a whole world of circular dependencies. 
Maybe even more difficult would be this one:
<img width="325" alt="Scherm­afbeelding 2022-08-24 om 15 04 48" src="https://user-images.githubusercontent.com/48658420/186425747-35a1f598-d481-4d0e-959f-a9b79dce9863.png">
In this case, I would expect the last note to be moved to the right to avoid the collision between the accidental and the slur. I guess that the difficulty is that slurs can only be laid out after horizontal spacing is fixed, but to solve a problem like this, the slurs would need to influence the horizontal spacing, so everything needs to be done again...